packages/cross_file/lib/src/types/io.dart

```diff
@@ -82,9 +92,17 @@ class XFile extends XFileBase {
   @override
   String get path => _file.path;
 
+  /// the name extracted from the last part of [File.path].
+  ///
+  /// **Be aware**, the name here is not the one
+  /// you may have passed in the constructor/factory.
```

Comment:
The client of this API is often not the person who called the constructor, which makes this comment confusing. I think the comment you've add to the constructor is sufficient.